lib/require_dependency_backward_compatibility.rb

```diff
@@ -18,5 +18,5 @@ def require_dependency(filename)
     super
   end
 
-  ActiveSupport::Dependencies::ZeitwerkIntegration::RequireDependency.prepend(self)
+  Object.prepend(self)
```

Comment:
Do we have to modify `Object` here? `ActiveSupport::Dependencies::RequireDependency` won't work?

Reply:
Yep, in Rails 7+ the code changed: https://github.com/rails/rails/blob/v7.0.2.3/activesupport/lib/active_support/dependencies/require_dependency.rb
Just tried out to check and patching the AS module won’t work.